### ISSUE TITLE
issues with statements of the form allocate ( A :: B)

### ISSUE THREAD

**Issue by eikehmueller** [stfc/fparser#76]
I've got a file which contains the following line:

```allocate( bicgstab_type :: gw_pressure_solver )```

This crashes fparser with the error below, there seems to be an issue with statements of the form ```allocate( A :: B)```. Note that if I do *not* run the file through fparser the line doesn't cause any issues and the code compiles. Any ideas what could cause this issue? Thanks, Eike

```
Full psyclone working/gungho/algorithm/gravity_wave_alg_mod.x90
Traceback (most recent call last):
  File "/home/n/em459/scratch/library//psyclone/bin/psyclone", line 4, in <module>
    __import__('pkg_resources').run_script('PSyclone==1.5.1', 'psyclone')
  File "build/bdist.linux-x86_64/egg/pkg_resources/__init__.py", line 735, in run_script
  File "build/bdist.linux-x86_64/egg/pkg_resources/__init__.py", line 1652, in run_script
  File "/beegfs/scratch/user/n/em459/library/psyclone/lib/python2.7/site-packages/PSyclone-1.5.1-py2.7.egg/EGG-INFO/scripts/psyclone", line 42, in <module>
    main(sys.argv[1:])
  File "/home/n/em459/scratch/library/psyclone/lib/python2.7/site-packages/PSyclone-1.5.1-py2.7.egg/psyclone/generator.py", line 228, in main
    alg_str = fll.process(str(alg))
  File "/home/n/em459/scratch/library/fparser-0.0.6/lib/python2.7/site-packages/fparser-0.0.6-py2.7.egg/fparser/base_classes.py", line 571, in __str__
    return self.tofortran()
  File "/home/n/em459/scratch/library/fparser-0.0.6/lib/python2.7/site-packages/fparser-0.0.6-py2.7.egg/fparser/block_statements.py", line 268, in tofortran
    return tab + BeginStatement.tofortran(self, isfix=isfix)
  File "/home/n/em459/scratch/library/fparser-0.0.6/lib/python2.7/site-packages/fparser-0.0.6-py2.7.egg/fparser/base_classes.py", line 687, in tofortran
    l.append(c.tofortran(isfix=isfix))
  File "/home/n/em459/scratch/library/fparser-0.0.6/lib/python2.7/site-packages/fparser-0.0.6-py2.7.egg/fparser/base_classes.py", line 687, in tofortran
    l.append(c.tofortran(isfix=isfix))
  File "/home/n/em459/scratch/library/fparser-0.0.6/lib/python2.7/site-packages/fparser-0.0.6-py2.7.egg/fparser/base_classes.py", line 687, in tofortran
    l.append(c.tofortran(isfix=isfix))
  File "/home/n/em459/scratch/library/fparser-0.0.6/lib/python2.7/site-packages/fparser-0.0.6-py2.7.egg/fparser/statements.py", line 671, in tofortran
    t = self.spec.tostr() + ' :: '
AttributeError: 'str' object has no attribute 'tostr'
```

**Comment by arporter** [comment]
This is a problem with `statement.Allocate.tofortran()`.

**Comment by TeranIvy** [comment]
Merged PR #77 to master and deleted the branch. Closing issue.
